### DISCUSSION TITLE
(Capstone DA) On creating my own plot, I chose boxplot and the outcome looks weird.

### DISCUSSION THREAD

**Discussion by mynugraErlangga** [teamalgoritma/community#661] [Q&A] [answered]
Good evening
I am attempting to answer the last question on creating my own plot. Below are the steps I took:
Saya ingin menjawab pertanyaan terakhir. Langkah2 sbb:
1. Subset to get a dataframe that only shows four select-categories
![Screenshot 2023-05-15 004925](https://github.com/teamalgoritma/community/assets/125937571/a7b653df-9219-4ebf-aabb-13851ff5ed61)

2. Aggregate using crosstab to list only four top categories based on 'Reviews'
![Screenshot 2023-05-15 005021](https://github.com/teamalgoritma/community/assets/125937571/415cdd11-44d6-4415-8339-b24c0f5cc35a)

3. Attempt at coding to get a boxplot
![Screenshot 2023-05-15 005102](https://github.com/teamalgoritma/community/assets/125937571/0e87549b-027e-480d-a785-5b78ab0fd228)

Purpose: I want to have a boxplot depicting the apps with the sum of 'Reviews' on the y-axis and is grouped into those 4 separate categories (i.e 4 boxes)

Question: 1. Please help with the full boxplot codes to input into the app.py file
2. The boxplot on the above scshot seemed to show ALL categories on the x-axis; Are my steps correct so far?

Terima kasih :)

**Comment by rusdipermana** [comment, marked as answer]
Hi,  here's an example code you can use for the boxplot :
# group the data by category
```
category_grouped = playstore.groupby('Category')['Reviews'].sum()
```
# create boxplot
```
plt.figure(figsize=(8,6))
plt.boxplot([playstore[playstore['Category'] == cat]['Reviews'] for cat in category_grouped.index], labels=category_grouped.index)
```

# set plot title and axis labels
```
plt.title('Total Reviews of Apps by Category')
plt.xlabel('Category')
plt.ylabel('Total Reviews')
```
# rotate x-axis labels
```
plt.xticks(rotation=90)
```
# save plot as image
```
plt.savefig('reviews_category_boxplot.png', bbox_inches="tight")
```

 In the code above, we first group the data by app category using groupby() and sum() on the 'Reviews' column. The result is stored in the category_grouped variable.
Next, we create the boxplot using the plt.boxplot() function and pass a list of lists containing the data for each category as a parameter. The category labels are taken from the index of the category_grouped variable.
After creating the boxplot, we add a title and axis labels using the plt.title(), plt.xlabel(), and plt.ylabel() functions. Finally, we save the plot as an image using plt.savefig().

![output](https://github.com/teamalgoritma/community/assets/6158006/f7bdd43b-c116-47cb-95c6-25f27411a11b)
